[CI] TopNOperatorTests testRandomMultiValuesTopN failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/w2ja5aufr7rya/tests/:x-pack:plugin:esql:compute:test/org.elasticsearch.compute.operator.topn.TopNOperatorTests/testRandomMultiValuesTopN


**Reproduction line:**
```
./gradlew ':x-pack:plugin:esql:compute:test' --tests "org.elasticsearch.compute.operator.topn.TopNOperatorTests.testRandomMultiValuesTopN" -Dtests.seed=B6A71FE479B8CD48 -Dtests.locale=en-CA -Dtests.timezone=Pacific/Wallis -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
[Failure dashboard for `org.elasticsearch.compute.operator.topn.TopNOperatorTests#testRandomMultiValuesTopN`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testRandomMultiValuesTopN'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.compute.operator.topn.TopNOperatorTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
java.lang.AssertionError: Expected a list containing
 0: a list containing
   0: a list containing
     0: <-2.4924623E38F>
   1: a list containing
     0: <-1966252967>
   2: a list containing
     0: <-1875932930>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
 1: a list containing
   0: a list containing
     0: <-2.1481112E38F>
   1: a list containing
     0: <417064179>
   2: a list containing
     0: <1304973021>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
 2: a list containing
   0: a list containing
     0: <-1.0014058E38F>
   1: a list containing
     0: <-1629155104>
   2: a list containing
     0: <2103660704>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
 3: a list containing
   0: a list containing
     0: <-6.486994E37F>
   1: a list containing
     0: <-1740869602>
   2: a list containing
     0: <-775296488>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
 4: a list containing
   0: a list containing
     0: <-1.9044326E37F>
   1: a list containing
     0: <944087735>
   2: a list containing
     0: <1934362755>
   3: a list containing
     0: <false>
   4: a list containing
     0: null
 5: a list containing
   0: a list containing
     0: <-1.6622808E37F>
   1: a list containing
     0: <-1500106667>
   2: a list containing
     0: <-940881369>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
 6: a list containing
   0: a list containing
     0: <-1.6537464E37F>
   1: a list containing
     0: <98490671>
   2: a list containing
     0: <517735224>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
 7: a list containing
   0: a list containing
     0: <-4.6762493E36F>
   1: a list containing
     0: <-1226964911>
   2: a list containing
     0: <2111802760>
   3: a list containing
     0: <false>
   4: a list containing
     0: null
 8: a list containing
   0: a list containing
     0: <-4.829915E35F>
   1: a list containing
     0: <-1038348581>
   2: a list containing
     0: <1392432291>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
 9: a list containing
   0: a list containing
     0: <-1.4362174E35F>
   1: a list containing
     0: <-1963738061>
   2: a list containing
     0: <652576908>
   3: a list containing
     0: <false>
   4: a list containing
     0: null
10: a list containing
   0: a list containing
     0: <-4.4533887E33F>
   1: a list containing
     0: <1309165482>
   2: a list containing
     0: <1493435023>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
11: a list containing
   0: a list containing
     0: <-2.7847876E33F>
   1: a list containing
     0: <-1907939981>
   2: a list containing
     0: <-12117310>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
12: a list containing
   0: a list containing
     0: <-2.4107301E33F>
   1: a list containing
     0: <-1404980140>
   2: a list containing
     0: <575190713>
   3: a list containing
     0: <false>
   4: a list containing
     0: null
13: a list containing
   0: a list containing
     0: <-3.4664565E32F>
   1: a list containing
     0: <-2083179184>
   2: a list containing
     0: <1752668719>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
14: a list containing
   0: a list containing
     0: <-3.076286E32F>
   1: a list containing
     0: <-1690925683>
   2: a list containing
     0: <1531909788>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
15: a list containing
   0: a list containing
     0: <-2.7993254E32F>
   1: a list containing
     0: <-117882321>
   2: a list containing
     0: <1352261033>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
16: a list containing
   0: a list containing
     0: <-2.5920542E32F>
   1: a list containing
     0: <-1999117702>
   2: a list containing
     0: <1528579622>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
17: a list containing
   0: a list containing
     0: <-1.9743132E32F>
   1: a list containing
     0: <-1483520812>
   2: a list containing
     0: <1800568218>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
18: a list containing
   0: a list containing
     0: <-1.2641199E32F>
   1: a list containing
     0: <-1898076629>
   2: a list containing
     0: <-1379953241>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
19: a list containing
   0: a list containing
     0: <-4.1077757E31F>
   1: a list containing
     0: <880209444>
   2: a list containing
     0: <2138463795>
   3: a list containing
     0: <false>
   4: a list containing
     0: null
20: a list containing
   0: a list containing
     0: <-4.0628157E31F>
   1: a list containing
     0: <422700328>
   2: a list containing
     0: <497953601>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
21: a list containing
   0: a list containing
     0: <-2.760985E31F>
   1: a list containing
     0: <618403570>
   2: a list containing
     0: <1606140700>
   3: a list containing
     0: <false>
   4: a list containing
     0: null
22: a list containing
   0: a list containing
     0: <-1.7803224E31F>
   1: a list containing
     0: <914623255>
   2: a list containing
     0: <-114212172>
   3: a list containing
     0: <false>
   4: a list containing
     0: null
23: a list containing
   0: a list containing
     0: <-4.3714344E30F>
   1: a list containing
     0: <-1884362663>
   2: a list containing
     0: <1628340971>
   3: a list containing
     0: <false>
   4: a list containing
     0: null
24: a list containing
   0: a list containing
     0: <-3.7201032E30F>
   1: a list containing
     0: <1109607945>
   2: a list containing
     0: <-566427415>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
25: a list containing
   0: a list containing
     0: <-1.8397557E28F>
   1: a list containing
     0: <-1806574384>
   2: a list containing
     0: <623665674>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
26: a list containing
   0: a list containing
     0: <-1.4880838E28F>
   1: a list containing
     0: <-506383244>
   2: a list containing
     0: <2103660704>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
27: a list containing
   0: a list containing
     0: <-1.8066809E27F>
   1: a list containing
     0: <-2069279624>
   2: a list containing
     0: <-2080288901>
   3: a list containing
     0: <false>
   4: a list containing
     0: null
28: a list containing
   0: a list containing
     0: <-8.03084E26F>
   1: a list containing
     0: <-1484085020>
   2: a list containing
     0: <1556886331>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
29: a list containing
   0: a list containing
     0: <-2.4445024E26F>
   1: a list containing
     0: <-1905907047>
   2: a list containing
     0: <1834707748>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
30: a list containing
   0: a list containing
     0: <-1.18618826E24F>
   1: a list containing
     0: <-1811179096>
   2: a list containing
     0: <1817106019>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
31: a list containing
   0: a list containing
     0: <-6.965261E21F>
   1: a list containing
     0: <-650694629>
   2: a list containing
     0: <-856248159>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
32: a list containing
   0: a list containing
     0: <-4.766824E21F>
   1: a list containing
     0: <-1439464381>
   2: a list containing
     0: <-1542840582>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
33: a list containing
   0: a list containing
     0: <-3.560909E21F>
   1: a list containing
     0: <-2048570785>
   2: a list containing
     0: <973697100>
   3: a list containing
     0: <false>
   4: a list containing
     0: null
34: a list containing
   0: a list containing
     0: <-1.1446563E21F>
   1: a list containing
     0: <883693534>
   2: a list containing
     0: <1739587089>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
35: a list containing
   0: a list containing
     0: <-1.7535596E19F>
   1: a list containing
     0: <1190204815>
   2: a list containing
     0: <372292485>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
36: a list containing
   0: a list containing
     0: <-5.0937845E17F>
   1: a list containing
     0: <1895658984>
   2: a list containing
     0: <1113962094>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
37: a list containing
   0: a list containing
     0: expected <-3.1150133E15F> but was <-8.2755636E12F>
   1: a list containing
     0: expected <59024213> but was <-1966199484>
   2: a list containing
     0: expected <2075463078> but was <-1166885856>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
38: a list containing
   0: a list containing
     0: expected <-8.2755636E12F> but was <-3.1531076E12F>
   1: a list containing
     0: expected <-1966199484> but was <-1131578473>
   2: a list containing
     0: expected <-1166885856> but was <572840338>
   3: a list containing
     0: expected <true> but was <false>
   4: a list containing
     0: null
39: a list containing
   0: a list containing
     0: expected <-3.1531076E12F> but was <-5.573515E9F>
   1: a list containing
     0: expected <-1131578473> but was <-1974158715>
   2: a list containing
     0: expected <572840338> but was <854205649>
   3: a list containing
     0: expected <false> but was <true>
   4: a list containing
     0: null
40: a list containing
   0: a list containing
     0: expected <-5.573515E9F> but was <-199474.33F>
   1: a list containing
     0: expected <-1974158715> but was <-1935555009>
   2: a list containing
     0: expected <854205649> but was <-271053814>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
41: a list containing
   0: a list containing
     0: expected <-199474.33F> but was <-116345.99F>
   1: a list containing
     0: expected <-1935555009> but was <-1577542898>
   2: a list containing
     0: expected <-271053814> but was <-1756368164>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
42: a list containing
   0: a list containing
     0: expected <-116345.99F> but was <-47690.418F>
   1: a list containing
     0: expected <-1577542898> but was <-555256913>
   2: a list containing
     0: expected <-1756368164> but was <652419948>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
43: a list containing
   0: a list containing
     0: expected <-47690.418F> but was <-1.8135599F>
   1: a list containing
     0: expected <-555256913> but was <-2109478332>
   2: a list containing
     0: expected <652419948> but was <1751365348>
   3: a list containing
     0: expected <true> but was <false>
   4: a list containing
     0: null
44: a list containing
   0: a list containing
     0: expected <-1.8135599F> but was <-0.7335181F>
   1: a list containing
     0: expected <-2109478332> but was <-1837447034>
   2: a list containing
     0: expected <1751365348> but was <-1696267267>
   3: a list containing
     0: expected <false> but was <true>
   4: a list containing
     0: null
45: a list containing
   0: a list containing
     0: expected <-0.7335181F> but was <-0.029040612F>
   1: a list containing
     0: expected <-1837447034> but was <-1029494035>
   2: a list containing
     0: expected <-1696267267> but was <1788882900>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
46: a list containing
   0: a list containing
     0: expected <-0.029040612F> but was <-0.00230249F>
   1: a list containing
     0: expected <-1029494035> but was <-1663945794>
   2: a list containing
     0: expected <1788882900> but was <494045317>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
47: a list containing
   0: a list containing
     0: expected <-0.00230249F> but was <-1.50973E-4F>
   1: a list containing
     0: expected <-1663945794> but was <-1959337067>
   2: a list containing
     0: expected <494045317> but was <-1352945150>
   3: a list containing
     0: <true>
   4: a list containing
     0: null
48: a list containing
   0: a list containing
     0: expected <-1.50973E-4F> but was <-8.6132895E-6F>
   1: a list containing
     0: expected <-1959337067> but was <1831427133>
   2: a list containing
     0: expected <-1352945150> but was <1953878181>
   3: a list containing
     0: expected <true> but was <false>
   4: a list containing
     0: null

  at __randomizedtesting.SeedInfo.seed([B6A71FE479B8CD48:E5E15A8D47FF8420]:0)
  at org.elasticsearch.test.MapMatcher.assertMap(MapMatcher.java:82)
  at org.elasticsearch.test.MapMatcher.assertMap(MapMatcher.java:65)
  at org.elasticsearch.compute.operator.topn.TopNOperatorTests.testRandomMultiValuesTopN(TopNOperatorTests.java:1099)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at org.junit.rules.RunRules.evaluate(RunRules.java:20)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1570)

```

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -75,9 +75,6 @@ tests:
   method: "testFetchAllEntities"
 - class: "org.elasticsearch.xpack.ml.integration.AutodetectMemoryLimitIT"
   issue: "https://github.com/elastic/elasticsearch/issues/109904"
-- class: "org.elasticsearch.compute.operator.topn.TopNOperatorTests"
-  issue: "https://github.com/elastic/elasticsearch/issues/109915"
-  method: "testRandomMultiValuesTopN"
 
 
 # Examples:

--- a/x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/BlockTestUtils.java
+++ b/x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/BlockTestUtils.java
@@ -17,6 +17,7 @@ import static org.elasticsearch.compute.data.BlockUtils.toJavaObject;
 import static org.elasticsearch.test.ESTestCase.between;
 import static org.elasticsearch.test.ESTestCase.randomBoolean;
 import static org.elasticsearch.test.ESTestCase.randomDouble;
+import static org.elasticsearch.test.ESTestCase.randomFloat;
 import static org.elasticsearch.test.ESTestCase.randomInt;
 import static org.elasticsearch.test.ESTestCase.randomLong;
 import static org.elasticsearch.test.ESTestCase.randomRealisticUnicodeOfCodepointLengthBetween;
@@ -31,7 +32,7 @@ public class BlockTestUtils {
         return switch (e) {
             case INT -> randomInt();
             case LONG -> randomLong();
-            case FLOAT -> Float.intBitsToFloat(randomInt());
+            case FLOAT -> randomFloat();
             case DOUBLE -> randomDouble();
             case BYTES_REF -> new BytesRef(randomRealisticUnicodeOfCodepointLengthBetween(0, 5));   // TODO: also test spatial WKB
             case BOOLEAN -> randomBoolean();

